src/ast/Node.hpp

```diff
@@ -196,6 +220,26 @@ const T& Node::get() const {
 	return boost::get<T>(*storagePtr);
 }
 
+template<class T, class U, class... Ts>
```

Comment:
Why do you need U? `<class T, class... Ts>` should be enough. Or even just <class... Ts>.